tilingManager/tilingLayouts/maximize.js

```diff
@@ -32,43 +34,74 @@ var MaximizeLayout = class MaximizeLayout extends BaseTilingLayout {
     }
 
     onTileRegulars(windows) {
-        if (this.animationInProgress) return;
+        super.onTileRegulars(windows);
+        const workArea = this.getWorkspaceBounds();
 
         windows.forEach(window => {
-            if (window.grabbed) return;
-            if (!window.maximized_horizontally) {
-                window.maximize(Meta.MaximizeFlags.BOTH);
```

Comment:
I prefer to keep maximized mode to maximazed layout to force windows that don't want to resize properly ( Like gnome-terminal) to be displayed correclty

Reply:
If we keep the maximized status, the windows don't have a correct size when drag and dropping

Reply:
And we can't force it because it seem to be sized by gnome-shell (mutter?) on unmaximize

Reply:
I restored it for now